integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/freeipa/FreeIpaUpgradeTests.java

```diff
@@ -301,10 +308,40 @@ private void kinit(SdxTestDto sdxTestDto, SdxClient sdxClient) {
         }
     }
 
+    private void dnsLookups(SdxTestDto sdxTestDto, SdxClient sdxClient) {
+        String fqdn = null;
+        String privateIP = "";
+        List<InstanceGroupV4Response> instanceGroups = getInstanceGroups(sdxTestDto, sdxClient);
+        InstanceMetaDataV4Response instanceGroupMetadata = instanceGroups.stream()
+                .flatMap(instanceGroup -> instanceGroup.getMetadata().stream())
+                .filter(metadata -> metadata.getInstanceGroup().equals("idbroker")).iterator().next();
+        if (instanceGroupMetadata == null) {
+            throw new TestFailException("FreeIPA upgrade DNS lookups test failed, idbroker instance group was not found.");
+        }
+        fqdn = instanceGroupMetadata.getDiscoveryFQDN();
+        privateIP = instanceGroupMetadata.getPrivateIp();
+        try {
+            String cmd = String.format(CHECK_DNS_LOOKUPS_CMD, fqdn, privateIP);
+            Map<String, Pair<Integer, String>> result = sshJClientActions.executeSshCommand(getInstanceGroups(sdxTestDto, sdxClient),
+                    List.of(HostGroupType.MASTER.getName()), cmd, false);
+            Iterator<Map.Entry<String, Pair<Integer, String>>> resultIterator = result.entrySet().iterator();
+            while (resultIterator.hasNext()) {
+                String status = resultIterator.next().getValue().getRight();
+                Assertions.assertTrue(status.contains("3"));
+                resultIterator.remove();
+            }
```

Comment:
this code is still a bit bloated:
```
        InstanceMetaDataV4Response instanceGroupMetadata = getInstanceGroups(sdxTestDto, sdxClient).stream()
                .flatMap(instanceGroup -> instanceGroup.getMetadata().stream())
                .filter(metadata -> metadata.getInstanceGroup().equals("idbroker"))
                .filter(metadata -> StringUtils.isNoneBlank(metadata.getDiscoveryFQDN(), metadata.getPrivateIp()))
                .findFirst().orElseThrow(() -> new TestFailException("FreeIPA upgrade DNS lookups test failed, idbroker instance group was not found."));
        try {
            String cmd = String.format(CHECK_DNS_LOOKUPS_CMD, instanceGroupMetadata.getDiscoveryFQDN(), instanceGroupMetadata.getPrivateIp());
            Map<String, Pair<Integer, String>> results = sshJClientActions.executeSshCommand(getInstanceGroups(sdxTestDto, sdxClient),
                    List.of(HostGroupType.MASTER.getName()), cmd, false);
            results.values().forEach(result -> Assertions.assertTrue(result.getRight().contains("3")));
```
I'm not sure why is it good that we count the IP's on ssh side and then assert on that value. I think the only important thing is if the IP is in the result.
`grep -q $IP` would return with `0` if it's present, and a different value if not. So I would either use this to validate, or bring back the result of the ping to assert if it contains the IP.

Reply:
Done and updated the PR.